### PR TITLE
Update TokenListController tests to not create networkController inst…

### PR DESCRIPTION
**Pull request by BelfordZ**
**No NetworkController in tests**

**Description**:
Mocking the behaviour of the network controller makes the tests more flexible, and lessens the burden of cleanup / maintainability of the tests. This PR removes the `NetworkController` from the `TokenListController` tests. 

Relates to #1051

### CODE REVIEW

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: packages/assets-controllers/src/TokenListController.test.ts [hunk 1]
```diff
@@ -1211,7 +1126,11 @@ describe('TokenListController', () => {
         resolve();
       });
 
-      network.setRpcTarget('http://localhost', '56');
+      controllerMessenger.publish('NetworkController:providerConfigChange', {
```
Comment (Gudahtt) [inline packages/assets-controllers/src/TokenListController.test.ts:1129]: Great to see more examples of how to use the top-level controller messenger in tests to simulate events. Not enough examples of this in our tests, I was asked about how to do this a few times now.

File: packages/assets-controllers/src/TokenListController.test.ts [hunk 2]
```diff
@@ -638,16 +608,11 @@ describe('TokenListController', () => {
       .persist();
 
     const controllerMessenger = getControllerMessenger();
-    const { network } = setupNetworkController(controllerMessenger);
     const messenger = getRestrictedMessenger(controllerMessenger);
-
+    let onNetworkStateChangeCallback: any;
```
Comment (mcmire) [inline packages/assets-controllers/src/TokenListController.test.ts:612]: What do you think about providing the type here instead of using `any`?

Comment (BelfordZ) [reply packages/assets-controllers/src/TokenListController.test.ts:612]: When i add the type: `let onNetworkStateChangeCallback: (state: NetworkState | ProviderConfig) => void;`

I get an error when trying to use it:
```
FAIL assets-controllers src/TokenListController.test.ts
➤ YN0000: [@metamask/assets-controllers]:   ● Test suite failed to run
➤ YN0000: [@metamask/assets-controllers]: 
➤ YN0000: [@metamask/assets-controllers]:     src/TokenListController.test.ts:625:5 - error TS2454: Variable 'onNetworkStateChangeCallback' is used before being assigned.
➤ YN0000: [@metamask/assets-controllers]: 
➤ YN0000: [@metamask/assets-controllers]:     625     onNetworkStateChangeCallback({ chainId: '5' });
````

Comment (BelfordZ) [reply packages/assets-controllers/src/TokenListController.test.ts:612]: resolved by changing to: 

`    let onNetworkStateChangeCallback!: (state: NetworkState | ProviderConfig) => void;`

File: packages/assets-controllers/src/TokenListController.test.ts [hunk 3]
```diff
@@ -657,14 +628,11 @@ describe('TokenListController', () => {
     expect(controller.state.tokenList).toStrictEqual(
       sampleSingleChainState.tokenList,
     );
-    network.setProviderType('goerli');
+    onNetworkStateChangeCallback({ chainId: '5' } as ProviderConfig);
```
Comment (Gudahtt) [inline packages/assets-controllers/src/TokenListController.test.ts:631]: No need to cast here; we just need one more property to make this a valid argument:
```suggestion
    onNetworkStateChangeCallback({
      chainId: '5',
      type: 'rpc',
    });
```

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by mcmire — APPROVED** [review]
Looks good!

File: packages/assets-controllers/src/TokenListController.test.ts (hunk 2 above)
Comment (mcmire) [reply packages/assets-controllers/src/TokenListController.test.ts]: Ah, nice, and I see why you had to add the `!` here. Is this documented anywhere in the TypeScript documentation to your knowledge? The only reference I could really find (at least for this particular usage) was here: https://devblogs.microsoft.com/typescript/announcing-typescript-2-7/#definite-assignment-assertions

**Pull request merged**